### PR TITLE
stm32: add more target boards to pwm samples application

### PR DESCRIPTION
**Pull request by FRASTM**
This adds the configuration to run tests and sample applications:

-  samples/basic/blinky_pwm/boards
-  samples/drivers/led_pwm
-  tests/drivers/pwm/pwm_api
 on the nucleo_l073rz and nucleo_f091rc plus the nucleo_g474re
All those 64-pin nucleo boards have the user led on the PA5 so that PWM must output on PA5 

Signed-off-by: Francois Ramu <francois.ramu@st.com>

### CODE REVIEW

**Comment by FRASTM** [comment]
- rebase on c32d10cc9e13d9e050d27b5804c0213f6cd3f452
- fix trailing space in .overlay

File: samples/basic/blinky_pwm/boards/nucleo_f091rc.overlay [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+		status = "okay";
+		pinctrl-0 = <&tim2_ch1_pa5>;
+		pinctrl-names = "default";
+	};
+};
+
+/* conflicting on pin PA5 */
+&spi1 {
+	status = "disabled";
+};
```
Comment (erwango) [inline samples/basic/blinky_pwm/boards/nucleo_f091rc.overlay:38]: Not required as this overlay is sample specific and sample doesn't enable `CONFIG_SPI`.

File: samples/basic/blinky_pwm/boards/nucleo_l073rz.overlay [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+		status = "okay";
+		pinctrl-0 = <&tim2_ch1_pa5>;
+		pinctrl-names = "default";
+	};
+};
+
+/* conflicting on pin PA5 */
+&spi1 {
+	status = "disabled";
+};
```
Comment (erwango) [inline samples/basic/blinky_pwm/boards/nucleo_l073rz.overlay:38]: Not required as this overlay is sample specific and sample doesn't enable `CONFIG_SPI`.

File: samples/drivers/led_pwm/boards/nucleo_f091rc.overlay [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+		status = "okay";
+		pinctrl-0 = <&tim2_ch1_pa5>;
+		pinctrl-names = "default";
+	};
+};
+
+/* conflicting on pin PA5 */
+&spi1 {
+	status = "disabled";
+};
```
Comment (erwango) [inline samples/drivers/led_pwm/boards/nucleo_f091rc.overlay:38]: Not required as this overlay is sample specific and sample doesn't enable `CONFIG_SPI`.

File: samples/drivers/led_pwm/boards/nucleo_l073rz.overlay [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+		status = "okay";
+		pinctrl-0 = <&tim2_ch1_pa5>;
+		pinctrl-names = "default";
+	};
+};
+
+/* conflicting on pin PA5 */
+&spi1 {
+	status = "disabled";
+};
```
Comment (erwango) [inline samples/drivers/led_pwm/boards/nucleo_l073rz.overlay:38]: Not required as this overlay is sample specific and sample doesn't enable `CONFIG_SPI`.

File: samples/basic/blinky_pwm/boards/nucleo_g474re.overlay [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+ *
+ * Copyright (c) 2022 STMicroelectronics
+ */
+
+#include <dt-bindings/pwm/pwm.h>
+
+&timers2 {
+	pwm2: pwm {
+		/delete-property/ pinctrl-0;
+		pinctrl-0 = <&tim2_ch1_pa5>;
```
Comment (erwango) [inline samples/basic/blinky_pwm/boards/nucleo_g474re.overlay:12]: Let's fix this directly in board.dts

**Review by erwango — CHANGES_REQUESTED** [review]

File: samples/drivers/led_pwm/boards/nucleo_l073rz.overlay (hunk 4 above)
Comment (FRASTM) [reply samples/drivers/led_pwm/boards/nucleo_l073rz.overlay]: remove this part of the spi1 node

File: samples/basic/blinky_pwm/boards/nucleo_f091rc.overlay (hunk 1 above)
Comment (FRASTM) [reply samples/basic/blinky_pwm/boards/nucleo_f091rc.overlay]: remove this part of the spi1 node

File: samples/basic/blinky_pwm/boards/nucleo_g474re.overlay (hunk 5 above)
Comment (FRASTM) [reply samples/basic/blinky_pwm/boards/nucleo_g474re.overlay]: change in the DTS of the nucleo_g474re

**Comment by FRASTM** [comment]
change DTS to remove SPI and nucleo_g474re

**Comment by FRASTM** [comment]
test results : 
```
tests/drivers/pwm/pwm_api/drivers.pwm,arm,nucleo_g474re,passed,,na,11.646256923675537,0,0
tests/drivers/pwm/pwm_api/drivers.pwm,arm,nucleo_f091rc,passed,,na,16.15660047531128,0,0
tests/drivers/pwm/pwm_api/drivers.pwm,arm,nucleo_l073rz,passed,,na,26.153932571411133,0,0
samples/drivers/led_pwm/sample.drivers.led.led_pwm,arm,nucleo_g474re,passed,,na,27.078551530838013,0,0
samples/drivers/led_pwm/sample.drivers.led.led_pwm,arm,nucleo_f091rc,passed,,na,23.28845238685608,0,0
samples/drivers/led_pwm/sample.drivers.led.led_pwm,arm,nucleo_l073rz,passed,,na,26.533900022506714,0,0
```

**Review by erwango — APPROVED** [review]

**Review by henrikbrixandersen — APPROVED** [review]

**Pull request merged**